### ISSUE TITLE
Unset values default to null

### ISSUE THREAD

**Issue by s314cy** [CodeDredd/pinia-orm#1769]
### Describe the feature

When defining a Pinia ORM model in TypeScript, optional fields must be typed as `field: type | null` because the `save` method of Pinia ORM repos converts unset values (null, undefined) to null.

However, the usual TypeScript way is declaring the optional field as `field?: type`, which treats an unset value as undefined. Furthermore, TypeScript's utility types also revolve around the undefined value with `Required<T>`, `Partial<T>`, etc.

My proposition is to replace the usage of null for unset values with undefined. This seems fairly easy and is rather a matter of acceptance by/relevance to the project.

### Additional information

- [X] Would you be willing to help implement this feature?

### Final checks

- [X] Check existing [discussions](https://github.com/CodeDredd/pinia-orm/discussions) and [issues](https://github.com/CodeDredd/pinia-orm/issues).

**Comment by s314cy** [comment]
Probably something to investigate: attributes default to null, but it seems that relations default to undefined 🤔

**Comment by CodeDredd** [comment]
I think you are right. But changing this, is something for v2 i think. Because it's a huge breaking change.
But i could already implement this behaviour with a feature flag.

**Comment by MartinX3** [comment]
The optional chaining works with `undefined` and `null`

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining

> If the object accessed or function called using this operator is [undefined](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/undefined) or [null](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/null), the expression short circuits and evaluates to [undefined](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/undefined) instead of throwing an error.

**Comment by s314cy** [comment]
> The optional chaining works with `undefined` and `null`
> 
> https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining
> 
> > If the object accessed or function called using this operator is [undefined](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/undefined) or [null](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/null), the expression short circuits and evaluates to [undefined](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/undefined) instead of throwing an error.

Indeed. But `foo?.bar` will return `undefined` if `foo` is `undefined`/`null`. Furthermore, consider the following code:

```ts
function fn<T>(arg?: T): unknown { ... }

class Foo extends Model {
  @Attr()
  declare bar: number | null;
}

const foo = useRepo(Foo).make();

// yields a compilation error because bar is null and not undefined
fn(foo.bar);
```

**Comment by CodeDredd** [comment]
Update on this one: it's still on the table, but it stays a v2.x opt-in — switching unset values from `null` to `undefined` by default breaks nearly every existing model definition, so it won't silently change in 2.0.0. The plan sketched here (a model/global config flag so `@Attr()` fields fill with `undefined`) is a good candidate right after 2.0.0 is stable. PRs welcome if you want to take a stab at it behind the flag.

**Comment by CodeDredd** [comment]
Closing as a design decision for 2.0.0: `null` stays the unset marker. Switching to `undefined` — even behind a flag — would create two behavioral worlds in docs, tests and support for a mostly ergonomic gain, and every serialization path (stores are JSON-serialized for SSR payloads) would silently drop `undefined` fields, which creates subtle hydration mismatches. Declaring fields as `field: type | null` remains the intended typing. I'll reconsider if there's strong demand after 2.0.0 settles.
